.github/workflows/build_deploy_docs.yml

```diff
@@ -18,6 +18,13 @@ jobs:
             5.0.x
             6.0.x
 
+      - uses: actions/cache@v3
+        with:
+          path: ~/.nuget/packages
+          key: ${{ runner.os }}-nuget-${{ hashFiles('**/*.csproj') }}
+          restore-keys: |
```

Comment:
Does it save cache ever? 🤔 Did is save it in past? 🤔 🤔 
My recent implementation of cache save/load: https://github.com/opensearch-project/sql-odbc/blob/2f43282431704e10b33de37a3d8046e2c98a339b/.github/workflows/sql-odbc-main.yml#L41-L61 (not a best example though)

Reply:
You can see it save here https://github.com/opensearch-project/opensearch-net/actions/runs/5502227861/jobs/10026345514?pr=265#step:8:4 (and restoring from an earlier cache here https://github.com/Xtansia/opensearch-net/actions/runs/5502227521/jobs/10026345103#step:4:17)

The only difference between this and the explicit restore/save steps, is that it doesn't try to save until after the workflow has fully succeeded, it doesn't save early.